### PR TITLE
conditionally load ES sniffer

### PR DESCRIPTION
**Pull request by konstan**
Depends on https://github.com/nuvla/api-server/pull/640 as this branch was created out of https://github.com/nuvla/api-server/tree/fix-459.

### CODE REVIEW

File: code/project.clj [hunk 1]
```diff
@@ -88,7 +88,10 @@
                                [org.apache.curator/curator-test]]
               :resource-paths ["test-resources"]
               :env            {:nuvla-session-key "test-resources/session.key"
-                               :nuvla-session-crt "test-resources/session.crt"}
+                               :nuvla-session-crt "test-resources/session.crt"
+                               :es-sniffer-init "no"
+                               :kafka-producer-init "yes"
+                               :kafka-client-conf-client-id "test-nuvla-server"}
```
Comment (0xbase12) [inline code/project.clj:94]: What is this client-id for ?

Comment (konstan) [reply code/project.clj:94]: When Kafka client (producer or consumer) contacts Kafka server, it needs to specify its ID (from docu: "The sole purpose of this is to be able to track the source of requests beyond just ip and port by allowing a logical application name to be included in Kafka logs and monitoring aggregates.") There is a default in the code which is "nuvla-server". Here I simply want to explicitly see in the logs that this is the test client and validate the propagation of the environment variables to the Kafka client.

File: code/src/sixsq/nuvla/server/util/kafka.clj [hunk 2]
```diff
@@ -1,12 +1,42 @@
...
+  "Given a map of env vars 'env-vars', return a map with Kafka only params
+  filtered out by 'prefix', and keys transformed into Kafka client params.
+  Example of the transformation:
+  Environment var: KAFKA_CLIENT_CONF_DELIVERY_TIMEOUT_MS=12345 ->
+  Clojure env var: {:kafka-client-conf-delivery-timeout-ms 12345} ->
+  Kafka parameter: {:delivery.timeout.ms 12345}"
+  [env-vars prefix]
+  (->> env-vars
+       (filter #(str/starts-with? (-> % first str) prefix))
+       (map #(identity [(-> %
```
Comment (0xbase12) [inline code/src/sixsq/nuvla/server/util/kafka.clj:32]: Small change, we don't need identity we can build a vector with following function
```
       (map (fn [[k v]]
              (vector (-> (str k)
                          (str/replace (re-pattern prefix) "")
                          (str/replace #"-" ".")
                          keyword) v)))
```

File: code/src/sixsq/nuvla/server/util/kafka.clj [hunk 3]
```diff
@@ -1,12 +1,42 @@
...
+                            (first)
+                            str
+                            (clojure.string/replace (re-pattern prefix) "")
```
Comment (0xbase12) [inline code/src/sixsq/nuvla/server/util/kafka.clj:35]: str namspace is already imported

File: code/src/sixsq/nuvla/server/util/kafka.clj [hunk 4]
```diff
@@ -1,12 +1,42 @@
...
+                            (clojure.string/replace #"-" ".")
+                            keyword) (second %)]))
```
Comment (0xbase12) [inline code/src/sixsq/nuvla/server/util/kafka.clj:37]: replace second with destructuring

**Review by 0xbase12 — APPROVED** [review]
1) Minor comments. 
2) I have some reformat because you have a difference in intellij you don't activated the align binidngs
3) ns import order fix
4) Merged master and fix changelog conflicts

**Pull request merged**